pkg/controller/windowsmachine/windows/windows.go

```diff
@@ -46,7 +48,7 @@ type Windows interface {
 	ID() string
 	// CopyFile copies the given file to the remote directory in the Windows VM. The remote directory is created if it
 	// does not exist
-	CopyFile(string, string) error
+	CopyFile(*files.FileInfo, string) error
```

Comment:
I feel as if the `CopyFile` name and description makes sense as this is an `Windows` interface signature. Skipping the actual copying if its unnecessary seems more like an implementation detail to me within the `windows` method. `CopyFile` makes more sense to me, but I can change it if we want to explicit about what is happening.

Reply:
I have to disagree here. `CopyFile` as it name states should always copy the file over whether if a file exists or not or if there is a checksum mismatch or not. You are actually changing the semantics of how `CopyFile` works with this change. Ideally `EnsureFile` should be added to the interface which should in turn call `CopyFile` only when needed. A change like this would be totally unacceptable if this was an exported library with other external callers.